.clang-format

```diff
@@ -1,56 +1,88 @@
-Cpp11BracedListStyle: true
-DerivePointerBinding: false
-IndentCaseLabels: false
-IndentFunctionDeclarationAfterType: false
-IndentWidth: 4
+﻿---
+AccessModifierOffset: '-2'
+AlignAfterOpenBracket: AlwaysBreak
+AlignConsecutiveMacros: 'true'
+AlignConsecutiveAssignments: 'true'
+AlignConsecutiveDeclarations: 'false'
+AlignEscapedNewlines: Right
+AlignOperands: 'false'
+AlignTrailingComments: 'false'
+AllowAllArgumentsOnNextLine: 'false'
+AllowAllConstructorInitializersOnNextLine: 'false'
+AllowAllParametersOfDeclarationOnNextLine: 'false'
+AllowShortBlocksOnASingleLine: 'false'
+AllowShortCaseLabelsOnASingleLine: 'false'
+AllowShortFunctionsOnASingleLine: Empty
+AllowShortIfStatementsOnASingleLine: Never
+AllowShortLambdasOnASingleLine: Empty
+AllowShortLoopsOnASingleLine: 'false'
+AlwaysBreakAfterDefinitionReturnType: None
+AlwaysBreakAfterReturnType: None
+AlwaysBreakBeforeMultilineStrings: 'false'
+AlwaysBreakTemplateDeclarations: 'Yes'
+BinPackArguments: 'false'
+BinPackParameters: 'false'
+BreakBeforeBraces: Attach
+BreakBeforeTernaryOperators: 'false'
+BreakConstructorInitializers: AfterColon
+BreakInheritanceList: AfterColon
+BreakStringLiterals: 'false'
+ColumnLimit: '115'
+CompactNamespaces: 'false'
+ConstructorInitializerAllOnOneLineOrOnePerLine: 'false'
+ConstructorInitializerIndentWidth: '4'
+ContinuationIndentWidth: '4'
+Cpp11BracedListStyle: 'true'
+DerivePointerAlignment: 'false'
+DisableFormat: 'false'
+FixNamespaceComments: 'true'
+IncludeBlocks: Regroup
+IndentCaseLabels: 'true'
+IndentPPDirectives: AfterHash
+IndentWidth: '4'
+IndentWrappedFunctionNames: 'true'
+JavaScriptQuotes: Double
+KeepEmptyLinesAtTheStartOfBlocks: 'false'
 Language: Cpp
-MaxEmptyLinesToKeep: 1
-NamespaceIndentation: None
-ObjCSpaceAfterProperty: true
-ObjCSpaceBeforeProtocolList: true
-PenaltyBreakBeforeFirstCallParameter: 120
-PenaltyBreakComment: 300
-PenaltyBreakFirstLessLess: 0
-PenaltyBreakString: 1000
-PenaltyExcessCharacter: 1000000
-PenaltyReturnTypeOnItsOwnLine: 20
-SpaceBeforeAssignmentOperators: true
-SpaceBeforeParens: Always
-SpaceInEmptyParentheses: false
-SpacesBeforeTrailingComments: 2
-SpacesInAngles: false
-SpacesInCStyleCastParentheses: false
-SpacesInContainerLiterals: false
-SpacesInParentheses: false
-SortIncludes: true
+MaxEmptyLinesToKeep: '1'
+NamespaceIndentation: Inner
+ObjCSpaceAfterProperty: 'true'
+ObjCSpaceBeforeProtocolList: 'true'
+PenaltyBreakBeforeFirstCallParameter: '120'
+PenaltyBreakComment: '300'
+PenaltyBreakFirstLessLess: '0'
+PenaltyBreakString: '1000'
+PenaltyExcessCharacter: '1000000'
+PenaltyReturnTypeOnItsOwnLine: '20'
+PointerAlignment: Middle
+ReflowComments: 'false'
+SortIncludes: 'true'
+SortUsingDeclarations: 'true'
+SpaceAfterCStyleCast: 'false'
+SpaceAfterLogicalNot: 'false'
+SpaceAfterTemplateKeyword: 'false'
+SpaceBeforeAssignmentOperators: 'true'
+SpaceBeforeCpp11BracedList: 'false'
+SpaceBeforeCtorInitializerColon: 'true'
+SpaceBeforeInheritanceColon: 'true'
+SpaceBeforeParens: ControlStatements
```

Comment:
I think it's easier to read when `SpaceBeforeParens` is `ControlStatements`:

```cpp
void f() {
  if (true) {
    f();
  }
}
```

versus when `SpaceBeforeParens` is `Always`:

```cpp
void f () {
  if (true) {
    f ();
  }
}
```

versus when `SpaceBeforeParens` is `Never`:

```cpp
void f() {
  if(true) {
    f();
  }
}
```

With `SpaceBeforeParens` being `ControlStatements` it is clear what a function call site or function definition is, versus some expression that needs parens to evaluate in an order the developer intends. With `SpaceBeforeParens` set to `ControlStatements`, it is more obvious that there is an expression in the `if` clause has to evaluated and differentiates it from other call sites. `ControlStatements` is the most easiest to read in my opinion.

Reply:
This one I agree with,  it would be a big change but it shouldn't change the line count very much, and have thought about changing it.